contracts/deploy/03-routers/03-gnosis-to-arb-router.ts

```diff
@@ -30,42 +40,63 @@ const deployRouter: DeployFunction = async (hre: HardhatRuntimeEnvironment) => {
+  const suffix = chainId === RouterChains.ETHEREUM_MAINNET ? "" : "Testnet";
+  const contractBaseName = "RouterGnosisToArb";
+  const deploymentName = `${contractBaseName}${suffix}`;
+
+  if (!(RouterChains[chainId] in paramsByChainId)) {
+    throw new Error(`Unsupported chain ID: ${chainId}`);
+  }
 
   // ----------------------------------------------------------------------------------------------
   const hardhatDeployer = async () => {
-    const veaOutbox = await deployments.get("VeaOutboxGnosisToArb" + (chainId === 100 ? "" : "Testnet"));
-    const veaInbox = await deployments.get("VeaInboxGnosisToArb" + (chainId === 100 ? "" : "Testnet"));
+    const [veaOutbox, veaInbox] = await Promise.all([
+      deployments.get(`VeaOutboxGnosisToArb${suffix}`),
+      deployments.get(`VeaInboxGnosisToArb${suffix}`),
+    ]);
 
-    const router = await deploy("RouterGnosisToArb" + (chainId === 100 ? "" : "Testnet"), {
+    const router = await deploy(deploymentName, {
       from: deployer,
-      contract: "RouterGnosisToArb",
+      contract: contractBaseName,
       args: [arbitrumBridge, amb, veaInbox.address, veaOutbox.address, chainId],
+      log: true,
     });
+
+    console.log(`Local deployment complete - ${deploymentName} deployed to: ${router.address}`);
   };
 
   // ----------------------------------------------------------------------------------------------
   const liveDeployer = async () => {
-    const outboxNetwork = chainId === 1 ? hre.companionNetworks.arbitrum : hre.companionNetworks.arbitrumSepolia;
-    const inboxNetwork = chainId === 1 ? hre.companionNetworks.gnosischain : hre.companionNetworks.chiado;
-    const veaOutbox = await outboxNetwork.deployments.get("VeaOutboxGnosisToArb" + (chainId === 1 ? "" : "Testnet"));
-    const veaInbox = await inboxNetwork.deployments.get("VeaInboxGnosisToArb" + (chainId === 1 ? "" : "Testnet"));
+    const { outboxNetwork, inboxNetwork } = getNetworkConfig(chainId);
 
-    const router = await deploy("RouterGnosisToArb" + (chainId === 1 ? "" : "Testnet"), {
-      from: deployer,
-      contract: "RouterGnosisToArb",
-      args: [arbitrumBridge, amb, veaInbox.address, veaOutbox.address, chainId],
-      log: true,
-    });
+    try {
+      const [veaOutbox, veaInbox] = await Promise.all([
+        hre.companionNetworks[outboxNetwork].deployments.get(`VeaOutboxGnosisToArb${suffix}`),
+        hre.companionNetworks[inboxNetwork].deployments.get(`VeaInboxGnosisToArb${suffix}`),
+      ]);
+
+      const router = await deploy(deploymentName, {
+        from: deployer,
+        contract: contractBaseName,
+        args: [arbitrumBridge, amb, veaInbox.address, veaOutbox.address, chainId],
+        log: true,
+      });
 
-    console.log("RouterGnosisToArb" + (chainId === 1 ? "" : "Testnet") + " deployed to: %s", router.address);
+      console.log(`${deploymentName} deployed to: ${router.address}`);
+    } catch (error) {
```

Comment:
This try/catch is not needed. Just let the error pop up, and you save yourself one execution branch that you will need to test in the future